### PR TITLE
Add GitHub action for test.

### PR DESCRIPTION
**Pull request by dkkb**
This workflow is inspired by one of my personal projects `dakv`. 

https://github.com/dakv/dakv/blob/master/.github/workflows/test.yml

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+on: [push, pull_request]
```
Comment (wu-sheng) [inline .github/workflows/test.yaml:1]: `push` should only be for the master branch.

**Comment by wu-sheng** [comment]
What is this actually testing?

**Comment by dkkb** [comment]
> What is this actually testing?

It can be used to run tests (include examples in documentation) with coverage and upload the code coverage reports to codecov.


image

File: .github/workflows/test.yaml (hunk 1 above)
Comment (dkkb) [reply .github/workflows/test.yaml]: Done.

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
Ping @Shikugawa for confirmation.

**Review by Shikugawa — APPROVED** [review]
Great. Thank you for your contribution!

**Pull request merged**